src/lib/rrd.lua

```diff
@@ -400,8 +400,9 @@ end
 
 local function compute_pdp_value(pdp, secs_per_pdp, dt, heartbeat_interval)
    -- This condition comes from upstream rrdtool.
-   if dt < heartbeat_interval and pdp.unknown_count < secs_per_pdp/2 then
-      return pdp.diff / tonumber(secs_per_pdp - pdp.unknown_count)
+   local unknown = tonumber(pdp.unknown_count)
```

Comment:
Pretty sure that as pdp is an FFI type (coming from one of the pdp_prep instances), pdp.unknown_count should always be a uint64_t.  WDYT?

Reply:
Ah I have been doing JS for too long :P  I see that this is a cast to a double, not a cast of not-a-number to a number.  I guess `<` wasn't working right comparing uint64 and doubles?  Was it casting doubles to uint64 or something?  Anyway cool :)

Reply:
I think the issue was that it is cast to an int which is fine for counter type RRDs (where the PDPs are integer values) but broke gauge type RRDs (PDPs are doubles.) Effectively IIRC the problem was truncation to zero for gauges with change rates of zero-point-something per second.